### PR TITLE
Program to run Filesystem stress on a disk (version 3)

### PR DESCRIPTION
**Pull request by hathyaga**
This program uses LTP to run fs related tests

Signed-off-by: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by hathyaga** [comment]
@PraveenPenguin @narasimhan-v 
New PR incorporated with suggested changes to ltp_fs.py script

File: io/disk/ltp_fs.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            self.skip("Please provide a disk for test")
+        self.mount_point = self.params.get('mount_point', default=self.srcdir)
+        self.script = self.params.get('script')
+        self.fs = self.params.get('fs', default='ext4')
+        self.args = self.params.get('args', default='')
+        self.log.info("creating %s filesystem on\
+                       disk %s" % (self.fs, self.disk))
+        cmd = "mkfs.%s %s" % (self.fs, self.disk)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("Creating filesystem %s on\
```
Comment (balamuruhans) [inline io/disk/ltp_fs.py]:
```
self.fail("Creating filesystem %s on"
          "%s failed" % (self.fs, self.disk))
```

File: io/disk/ltp_fs.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error(package + ' is needed for the test to be run')
+        self.disk = self.params.get('disk', default="")
+        if self.disk == "":
+            self.skip("Please provide a disk for test")
+        self.mount_point = self.params.get('mount_point', default=self.srcdir)
+        self.script = self.params.get('script')
+        self.fs = self.params.get('fs', default='ext4')
+        self.args = self.params.get('args', default='')
+        self.log.info("creating %s filesystem on\
```
Comment (balamuruhans) [inline io/disk/ltp_fs.py]: same here

File: io/disk/ltp_fs.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("Creating filesystem %s on\
+                       %s failed" % (self.fs, self.disk))
+        self.log.info("Mounting disk %s to mount point %s"
+                      % (self.disk, self.mount_point))
+        cmd = "mount %s %s" % (self.disk, self.mount_point)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("Unable to mount to location  %s" % self.mount_point)
+        url = "https://github.com/linux-test-project/ltp/\
+               archive/master.zip"
```
Comment (balamuruhans) [inline io/disk/ltp_fs.py]:
```
url = "https://github.com/linux-test-project/ltp/"
url += "archive/master.zip"
```

File: io/disk/ltp_fs.py [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        self.log.info("creating %s filesystem on\
+                       disk %s" % (self.fs, self.disk))
+        cmd = "mkfs.%s %s" % (self.fs, self.disk)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
+            self.fail("Creating filesystem %s on\
+                       %s failed" % (self.fs, self.disk))
+        self.log.info("Mounting disk %s to mount point %s"
+                      % (self.disk, self.mount_point))
+        cmd = "mount %s %s" % (self.disk, self.mount_point)
+        if process.system(cmd, shell=True, ignore_status=True) != 0:
```
Comment (balamuruhans) [inline io/disk/ltp_fs.py]: `if process.system(cmd, shell=True, ignore_status=True):` will be sufficient

File: io/disk/ltp_fs.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        ltpbin_dir = os.path.join(ltp_dir, 'bin')
+        os.mkdir(ltpbin_dir)
+        process.system('./configure --prefix=%s' % ltpbin_dir)
+        build.make(ltp_dir)
+        build.make(ltp_dir, extra_args='install')
+
+    def test_fs_run(self):
+
+        '''
+        Downloades LTP, complies, installs and runs filesystem
```
Comment (balamuruhans) [inline io/disk/ltp_fs.py]: %s/Downloades/Downloads

Comment (balamuruhans) [inline io/disk/ltp_fs.py]: %s/complies/compiles

File: io/disk/ltp_fs.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+            cmd = '%s %s' % (os.path.join(ltpbin_dir, self.script), self.args)
+            result = process.run(cmd, ignore_status=True)
+            # Walk the stdout and try detect failed tests from lines
+            # like these:
+            # aio01       5  TPASS  :  Test 5: 10 reads and
+            # writes in  0.000022 sec
+            # vhangup02    1  TFAIL  :  vhangup02.c:88:
+            # vhangup() failed, errno:1
+            # and check for fail_statuses The first part contain test name
+            fail_statuses = ['TFAIL', 'TBROK', 'TWARN']
```
Comment (balamuruhans) [inline io/disk/ltp_fs.py]: can have variable name as fail_status or fail_status_list

File: io/disk/ltp_fs.py [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+            ltpbin_dir = os.path.join(self.srcdir, "ltp-master", 'bin')
+            cmd = '%s %s' % (os.path.join(ltpbin_dir, self.script), self.args)
+            result = process.run(cmd, ignore_status=True)
+            # Walk the stdout and try detect failed tests from lines
+            # like these:
+            # aio01       5  TPASS  :  Test 5: 10 reads and
+            # writes in  0.000022 sec
+            # vhangup02    1  TFAIL  :  vhangup02.c:88:
+            # vhangup() failed, errno:1
+            # and check for fail_statuses The first part contain test name
```
Comment (balamuruhans) [inline io/disk/ltp_fs.py]: %s/fail_statuses/fail_status

File: io/disk/ltp_fs.py.data/README.txt [hunk 8]
```diff
@@ -0,0 +1,8 @@
+This program downloads the LTP test suite, compiles it and runs 2 filesystem related tests 'fs_di' and 'fs_inod' by involking the runltp test script.
```
Comment (balamuruhans) [inline io/disk/ltp_fs.py.data/README.txt]: %s/involking/invoking

File: io/disk/ltp_fs.py.data/ltp_fs.yaml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+            args: '-s fs_di'
+        test2:
+            args: '-s fs_inod'
+    fs: !mux
+        ext4:
+            fs: 'ext4'
+        ext3:
+           fs: 'ext3'
+        ext2:
+            fs: 'ext2'
```
Comment (narasimhan-v) [inline io/disk/ltp_fs.py.data/ltp_fs.yaml]: Nice. Thanks.

**Comment by hathyaga** [comment]
@narasimhan-v @balamuruhans  Closing this PR as a new PR is sent with changes suggested.
https://github.com/avocado-framework-tests/avocado-misc-tests/pull/226

**Pull request closed without merging**